spi/common/core-spi/src/main/java/org/eclipse/dataspaceconnector/spi/message/RemoteMessageDispatcher.java

```diff
@@ -36,6 +36,6 @@
      * @param context      the message context
      * @return a future that can be used to retrieve the response when the operation has completed
      */
-    <T> CompletableFuture<T> send(Class<T> responseType, RemoteMessage message, MessageContext context);
+    <T, M extends RemoteMessage> CompletableFuture<T> send(Class<T> responseType, M message, MessageContext context);
```

Comment:
`responseType` and `context` are unused. could we remove them?

Reply:
If we remove them from the dispatcher, they should probably also be removed from the dispatcher registry. `context` we can definitely remove. When removing `responseType`, we would lose the type of the completable future returned by the `send` method, meaning we'd always get back a `CompletableFuture<Object>`. Since we only have one implementation of `RemoteMessageDispatcher` at the moment, this shouldn't be a problem, but we may need to reconsider this when another protocol is added, to ensure that the expected response type is returned.

@paullatzelsperger what do you think?

Reply:
I think the return type can be obtained at runtime, so in fact the `responseType` currently does nothing, if we remove it, the dispatchment should work fine, in the `CatalogServiceImpl`, that's the only place where we actually bind a return type that's not `Object`, the type it's set from the method return type that's `CompletableFuture<Catalog>`, so at the end that will be the type returned. An eventual cast exception could be thrown in the `IdsMultipartRemoteMessageDispatcher` when the delegate is obtained, but that could happen also in the current main branch so I think it's ok to remove it.
Maybe it could be done in another PR to not put too much effort in this PR that brings already a lot of value.